docs/projects/reverse-a-string/_posts/2019-10-16-dart.md

```diff
@@ -0,0 +1,77 @@
+
+So let's look at that reverse function.  We've defined the function with type `String` as it'll return our reversed string to be printed to the console.  It takes the parameter `input` as an undefined variable but we're assuming here that it will be a string value.
+
+The `Dart:core` library String class contains the `split` method which will split a string or list of strings into a list of substrings based on the pattern given to it.  If the pattern is empty as in our reverse function, split will break up the input into single-code unit strings (individual characters).  So 'Hello' with the empty pattern `''` returns [H, e, l, l, o].
+
+After we've broken the string into characters with split, we'll use the `reversed` property from the `Dart:core` library's List class.  This property simply takes a list and returns it in reversed order as an Iterable object.  So our example list, [H, e, l, l, o], becomes (o, l, l, e, H).
+
+For the purposes of this sample program, there is very little difference between List and Iterable objects.  Both are in fact iterable.  At a high level, lists will have additional functionality such as indexed read/write access to its elements and sorting functions.  Iterables on the other hand are typically created once then accessed as readonly data during an iteration operation like a for-loop.
+
+Both Iterables and Lists have the `join` method which will take their elements and concatenate them into a string.  We use `join` here on the reversed Iterable which is returned as the final string value of our `reverse` function.  This value is then printed out to the console from `print` function within `main`.
+
+## How to Run Solution
+
+To run the Dart string reversal program, download the dart file from GitHub, install the Dart SDK as described at [dart.dev](https://dart.dev/get-dart), and run the following from the command line:
+
+```console
+dart reverse-string.dart "Hello, World!"
+```
+
+Alternatively, you can copy the source code into [DartPad](https://dartpad.dartlang.org/), an online Dart interpreter. Just keep in mind that you won't have access to the command line arguments input using this method so you'll have to populate the args variable in the code instead.  For example:
+
+```dart
+void main(List<String> args) {
+    args = ['Hello World'];
+    print( reverse(args[0]) );
+}
+String reverse(input) {
+    return input.split('').reversed.join();
+}
+```
+[^1]
+
+---
+
+#### References
+
+[^1]: P. Ryon (slashdoom), “Reverse a String in Dart,” The Renegade Coder, 16-Oct-2019. [Online]. Available: <https://therenegadecoder.com/code/reverse-a-string-in-dart/>. [Accessed: 16-Oct-2019].
```

Comment:
It looks like you're referencing an article that doesn't exist. Does that mean you'd like this article cross-posted on my personal blog? I'd be more than happy to do that!

Reply:
To be honest I just saw the links on so many of the other docs I though it must have been automatically cross-posted and went with it.  I'm happy to have this there as well though.  Thanks for that!

Reply:
Yeah, that's fine by me! Right now, it wouldn't get posted until November 1st. So, that link will be dead for about two weeks.